### DISCUSSION TITLE
Provide public telemetry statistics

### DISCUSSION THREAD

**Discussion by gjasny** [microsoft/vcpkg#17795] [Ideas]
Hello,

according to the [privacy.md](https://github.com/Microsoft/vcpkg/blob/master/docs/about/privacy.md)   document, vcpkg collects metrics about installed ports. It would be helpful if you could publish them in some aggregated and queryable form like for example [Debian Popularity Contest](https://popcon.debian.org/) does for Debian packages.

Thanks,
Gregor

**Comment by Mike-Devel** [comment]
I'd very much like to see those statistics too. If you are collecting our data without asking, the least you could do is show it to us.

Specifically what I would be interested in is which are the most commonly directly installed boost libraries - or rather if people install individual libraries at all or just boost.

**Comment by PhoebeHui** [comment]
This issue hasn’t been updated in a year; if it is still an issue, please reopen.

**Comment by gjasny** [comment]
@PhoebeHui please re-open and maybe escalate within Microsoft.

**Comment by PhoebeHui** [comment]
Thanks, reopen it.

**Comment by Mike-Devel** [comment]
> Thanks, reopen it.

Would be ideal, if it could also be addressed ;)

**Comment by JackBoosY** [comment]
That's really a good idea!

**Comment by ihsandemir** [comment]
Yes, please provide some statistics if possible.

**Comment by Mike-Devel** [comment]
So, what is your policy for moving issues to discussions?

**Reply by Mike-Devel** [reply]
@PhoebeHui

**Reply by PhoebeHui** [reply]
For some issues that bring up some good ideas or questions and still be valid, we would convert them for further discussion. 
See https://github.com/microsoft/vcpkg/discussions/17048.

**Comment by chausner** [comment]
Seeing aggregated statistics on the download numbers of ports would be really interesting. I have been focusing lately on updating outdated ports and have prioritized ports based on the number of stars on GitHub. If we had information about how popular libraries are on vcpkg, it would be easier to update those ports that affect the most people instead of going by their general popularity on GitHub.

**Reply by chausner** [reply]
Notice how Conan Center, for example, provides download numbers on their front page for packages: https://conan.io/center/.

**Reply by dg0yt** [reply]
This statistics will be biased by actual port usability. 
For example, "gdal" is fairly popular for geographic applications. However, the port turned out to be hardly usable on non-windows systems, and even now the static variant on windows is feature-incomplete. There is some evidence that people tested gdal in vcpkg, and turned away from vcpkg, both as users and as contributors. OTOH, we are now close to offer good cross-platform experience for gdal soon. This should shift numbers.
Briefly, ports are popular if they are already well mantained...

**Reply by kelteseth** [reply]
I have used @dg0yt Idea in https://arewemodulesyet.org/ that tracks C++ lib popularity and corresponding modules support. Checkout the script to generate the data at https://github.com/kelteseth/arewemodulesyet/blob/master/generate_vcpkg_usage_stats.py

![grafik](https://github.com/microsoft/vcpkg/assets/1071536/f7b72cfc-e82f-43e5-9e8e-9baedb05a04e)
